db/db_impl/db_impl.cc

```diff
@@ -5536,6 +5591,36 @@ Status DBImpl::GetCreationTimeOfOldestFile(uint64_t* creation_time) {
     return Status::NotSupported("This API only works if max_open_files = -1");
   }
 }
+
+void DBImpl::RecordSeqnoToTimeMapping() {
+  SequenceNumber min_first_mem_seqno = kMaxSequenceNumber;
+
+  for (auto cfd : *versions_->GetColumnFamilySet()) {
+    if (cfd->ioptions()->preclude_last_level_data_seconds > 0) {
+      min_first_mem_seqno =
+          std::min(cfd->GetFirstMemtableSequenceNumber(), min_first_mem_seqno);
```

Comment:
versions_->GetColumnFamilySet() and cfd->GetFirstMemtableSequenceNumber() seems to be called outside DB mutex. Is it safe?